libs/sdk-backend-spi/src/workspace/metadata/index.ts

```diff
@@ -58,4 +61,28 @@ export interface IWorkspaceMetadata {
      * @returns promise returning batch of attribute references.
      */
     getCommonAttributesBatch(attributesRefsBatch: ObjRef[][]): Promise<ObjRef[][]>;
+
+    /**
+     * Gets the list of metadata of attribute display form with the provided list of uris.
+     *
+     * @param uris - list of uris of display form.
+     * @returns promise of list of attribute display form metadata object.
+     */
+    getAttributeDisplayFormsByUris(uris: Uri[]): Promise<IAttributeDisplayFormMetadataObject[]>;
+
+    /**
+     * Gets the list of metadata of attribute with the provided list of uris.
+     *
+     * @param uris - list of uris of attribute.
+     * @returns promise of list of attribute metadata object.
+     */
+    getAttributesByUris(uris: Uri[]): Promise<IAttributeMetadataObject[]>;
+
+    /**
+     * Gets the list of metadata objects that contains only uri and identifier.
+     *
+     * @param identifiers -  list of identifiers of attribute.
+     * @returns promise of list of object.
+     */
+    getUrisByIdentifiers(identifiers: Identifier[]): Promise<IUriIdentifierObject[]>;
```

Comment:
functions like this are not acceptable on the SPI level. instead, you should use the opaque ObjRef (which can be either uri or id) and then have the backend handle them and do what is needed.

Reply:
I have updated the PR.